### ISSUE TITLE
ui5-middleware-serveframework + ui5-test-runner issue?

### ISSUE THREAD

**Issue by wozjac** [ArnaudBuchholz/ui5-test-runner#107]
Hi,

I noticed that test-runner fails if `ui5-middleware-serveframework` is used.

1. Go to https://github.com/SAP/openui5-sample-app
2. Add to `ui5.yaml`

```yaml
...
server:
  customMiddleware:
    - name: ui5-middleware-serveframework
      afterMiddleware: compression

    - name: "@ui5/middleware-code-coverage"
      afterMiddleware: compression
      configuration:
        excludePatterns:
          - "resources/"
          - "test/"
```

**Expected behavior**
Test runner executes.

**Report**
```sh
Died on test #1     at http://localhost:8080/resources/sap/ui/test/starter/runTest.js:1112:6559: Called start() while test already started running     
⠙ Generating reports
Error: Command failed with exit code 4294967295: npm run test-runner                                                                                   
    at makeError (C:\workspace\samples\openui5-sample-app\node_modules\execa\lib\error.js:60:11)
    at handlePromise (C:\workspace\samples\openui5-sample-app\node_modules\execa\index.js:118:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  shortMessage: 'Command failed with exit code 4294967295: npm run test-runner',
  command: 'npm run test-runner',
  escapedCommand: '"npm run test-runner"',
  exitCode: 4294967295,
```

BR,
Jacek

**Comment by ArnaudBuchholz** [comment]
Hello ! I was able to reproduce the issue. For some reasons, when the `ui5-middleware-serveframework` is being used, the tests are either starting too soon or twice. I am qualifying.

**Comment by ArnaudBuchholz** [comment]
Did some progress in my investigation : ui5-test-runner opens the pages with some preloaded scripts that hook important events during the loading of the tests. For some technical reasons, it seems like this generates bad interactions with the _compiled_ version of UI5 resources.

**Comment by ArnaudBuchholz** [comment]
Source of the problem : https://github.com/ArnaudBuchholz/ui5-test-runner/blob/84c0dce0d36a082b4558e6072bd49aaf57311450/src/inject/qunit-intercept.js

This code was created to handle asynchronous loading of QUnit : QUnit hooks might be registered *before* QUnit is actually loaded. This code defers the executions of the hooks but somewhat conflicts with the logic implemented in the new test runner.

**Comment by ArnaudBuchholz** [comment]
Hello ! I released version `5.3.7` that fixes the issue. can you please test and close the incident ?

**Comment by wozjac** [comment]
Worked, thanks for prompt reaction 👌
